### PR TITLE
add support for mongodb+srv URIs and TLS

### LINKED ISSUE #3971: Using using pyMongo srv tracker with latest Rasa docker image
Using latest rasa/rasa:latest-full docker image and getting the following error. Shouldn't the dnspython module come standard in the docker image? Or is this due to something else?:
```rasa_1           | 2019-07-10 04:06:20 INFO     root  - Starting Rasa Core server on http://localhost:5200
rasa_1           | 2019-07-10 04:06:20.650678: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
rasa_1           | 2019-07-10 04:06:21.221940: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2200000000 Hz
rasa_1           | 2019-07-10 04:06:21.277310: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x556f569865c0 executing computations on platform Host. Devices:
rasa_1           | 2019-07-10 04:06:21.277484: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
rasa_1           | [2019-07-10 04:06:22 +0000] [1] [ERROR] Experienced exception while trying to serve
rasa_1           | Traceback (most recent call last):
rasa_1           |   File "/usr/local/lib/python3.6/site-packages/sanic/app.py", line 1096, in run
rasa_1           |     serve(**server_settings)
rasa_1           |   File "/usr/local/lib/python3.6/site-packages/sanic/server.py", line 742, in serve
rasa_1           |     trigger_events(before_start, loop)
rasa_1           |   File "/usr/local/lib/python3.6/site-packages/sanic/server.py", line 604, in trigger_events
rasa_1           |     loop.run_until_complete(result)
rasa_1           |   File "uvloop/loop.pyx", line 1451, in uvloop.loop.Loop.run_until_complete
rasa_1           |   File "/usr/local/lib/python3.6/site-packages/rasa/core/run.py", line 176, in load_agent_on_start
rasa_1           |     None, endpoints.tracker_store, _broker
rasa_1           |   File "/usr/local/lib/python3.6/site-packages/rasa/core/tracker_store.py", line 38, in find_tracker_store
rasa_1           |     domain=domain, host=store.url, event_broker=event_broker, **store.kwargs
rasa_1           |   File "/usr/local/lib/python3.6/site-packages/rasa/core/tracker_store.py", line 205, in __init__
rasa_1           |     connect=False,
rasa_1           |   File "/usr/local/lib/python3.6/site-packages/pymongo/mongo_client.py", line 524, in __init__
rasa_1           |     res = uri_parser.parse_uri(entity, port, warn=True)
rasa_1           |   File "/usr/local/lib/python3.6/site-packages/pymongo/uri_parser.py", line 318, in parse_uri
rasa_1           |     raise ConfigurationError('The "dnspython" module must be '
rasa_1           | pymongo.errors.ConfigurationError: The "dnspython" module must be installed to use mongodb+srv:// URIs
```
**Content of endpoints file (endpoints.yml)**
```
action_endpoint:
  url: "http://localhost:5200/webhooks/twilio/webhook"

  tracker_store:
    store_type: mongod
    url: mongodb+srv://(etc.etc.etc.)
    db: [REDACTED]
    username: [REDACTED]
    password: [REDACTED]
```

relevant part of Docker-compose.yml:
```
version: '3.0'
  
services:
  rasa:
    image: rasa/rasa:latest-full
    networks: ['rasa-network']
    ports:
    - "5005:5005"
    volumes:
    - ./:/app
            #- "./models:/app/models"
            #- "./data:/app/data"
      #- "./rasa:/usr/local/lib/python3.6/site-packages/rasa"
    command: run --enable-api --model models --p 5200 --connector twilio --credentials credentials.yml --endpoints endpoints-twilio.yml
```

### PR DESCRIPTION
**Pull request by ortsaCniveK**
**Proposed changes**:
- adds support for `mongodb+srv://` URIs, simply by adding `dnspython`
    - Fixes #3971 and #2968  

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] updated the changelog

### CODE REVIEW

**Comment by imLew** [comment]
Awesome! Thank very much for this PR, I'll ask @wochinge for a review as he knows much more about Docker than I do.

File: requirements.txt [hunk 1]
```diff
@@ -7,6 +7,7 @@ attrs==19.1.0
 jsonpickle==1.1
 redis==3.2.1
 fakeredis==1.0.3
+dnspython==1.16.0
```
Comment (wochinge) [inline requirements.txt]: can we also do `pymongo[tls,srv]` as mentioned here https://stackoverflow.com/a/53644085/3429596 ?

**Comment by wochinge** [comment]
thanks for your pr @ortsaCniveK !

File: requirements.txt (hunk 1 above)
Comment (ortsaCniveK) [reply requirements.txt]: sure, I'll update the PR.

Comment (wochinge) [reply requirements.txt]: 🥇

**Comment by ortsaCniveK** [comment]
alright, it's updated @wochinge :smile:

**Comment by wochinge** [comment]
awesome! and a changelog update would be great! then we are done 🎈 🎉

**Comment by ortsaCniveK** [comment]
Added to 1.1.17, didn't quite make the cut for 1.1.16 :grin:

File: CHANGELOG.rst [hunk 2]
```diff
@@ -11,6 +11,7 @@ This project adheres to `Semantic Versioning`_ starting with version 1.0.
 
 Added
 -----
+- optional pymongo dependencies ``[tls, srv]`` to ``requirements.txt`` for better mongodb support
```
Comment (wochinge) [inline CHANGELOG.rst]:
```suggestion
- added optional pymongo dependencies ``[tls, srv]`` to ``requirements.txt`` for better mongodb support
```

**Comment by ortsaCniveK** [comment]
@wochinge Just double checked and was having trouble with connecting to mongodb using the new URIs, but it seems the `requirements.txt` changes were overwritten -- did you need me to update the PR?

**Comment by wochinge** [comment]
@ortsaCniveK Super weird, why this is not there any more 💥 Even checked git blame, but nothing there. @tmbo Any idea what happened here?

**Comment by tmbo** [comment]
Yes found it: regression got accidentally introduced in this PR https://github.com/RasaHQ/rasa/pull/4020/files - we need to readd to requirements.

**Comment by tmbo** [comment]
In any case, this should really be changed in setup.py as well

**Pull request merged**